src/vegur.erl

```diff
@@ -78,6 +81,17 @@ start_http(Port, Interface, Config) ->
     HttpRef = proplists:get_value(ref, Config, ?HTTP_REF),
     start(http, HttpRef, Port, Interface, Config).
 
+-spec start_https(PortNumber, Interface, Options, SSLOptions) ->
+                        {ok, pid()}|no_return() when
+      PortNumber :: inet:port_number(),
+      Interface :: module(),
+      Options :: options(),
+      SSLOptions :: [ssl:ssloption()].
+start_https(Port, Interface, Config, SSLOptions) ->
+    ok = set_config(extra_socket_options, SSLOptions ++ extra_socket_options()),
```

Comment:
Why is this required? 
I think there is a risk that this pollutes SSL socket options and uses them for non-SSL listeners depending on the start order.